src/skmultiflow/anomaly_detection/half_space_trees.py

```diff
@@ -0,0 +1,548 @@
+import random
+
+from sklearn.preprocessing import normalize
+from skmultiflow.core import BaseSKMObject, ClassifierMixin
+from skmultiflow.utils.utils import *
+
+
+class HalfSpaceTrees(BaseSKMObject, ClassifierMixin):
+    """
+    Half Space Trees
+
+    Parameters
+    ----------
+
+    dimensions: int, required
```

Comment:
Use `n_features` for consistency with the framework.

Reply:
I want to let you know that I have integrated all the small code refactoring.